piptools/repositories/pypi.py

```diff
@@ -201,6 +183,36 @@ def _get_file_hash(self, location):
                 h.update(chunk)
         return ":".join([FAVORITE_HASH, h.hexdigest()])
 
+    @contextmanager
+    def allow_all_wheels(self):
+        """
+        Monkey patches pip.Wheel to allow wheels from all platoforms and Python versions.
+
+        This also saves the candidate cache and set a new one, or else the results from the
+        previous non-patched calls will interfere.
+        """
+        def _wheel_supported(self, tags=None):
+            # Ignore current platform. Support everything.
+            return True
+
+        def _wheel_support_index_min(self, tags=None):
+            # All wheels are equal priority for sorting.
+            return 0
+
+        original_wheel_supported = Wheel.supported
+        original_support_index_min = Wheel.support_index_min
+        original_cache = self._available_candidates_cache
+
+        Wheel.supported = _wheel_supported
+        Wheel.support_index_min = _wheel_support_index_min
+        self._available_candidates_cache = {}
+
+        yield
```

Comment:
I would wrap this in a `try`/`finally` in case an exception is raised and then caught somewhere else in the stack.

```py
try:
    yield
finally:
     Wheel.supported = original_wheel_supported
     Wheel.support_index_min = original_support_index_min
     self._available_candidates_cache = original_cache
```

Reply:
Well I learned something today, I though the exiting part would always be called, even with an exception, without the `try`. Just tried it out and you're right, will fix this, thanks.